src/interactive/workspace.ts

```diff
@@ -1,45 +1,77 @@
 import * as vscode from 'vscode';
+import * as repl from './repl';
 
 let g_terminal: vscode.Terminal = null
-let g_replVariables: string = '';
 
-export class REPLTreeDataProvider implements vscode.TreeDataProvider<string> {
-    private _onDidChangeTreeData: vscode.EventEmitter<string | undefined> = new vscode.EventEmitter<string | undefined>();
-    readonly onDidChangeTreeData: vscode.Event<string | undefined> = this._onDidChangeTreeData.event;
+interface WorkspaceVariable {
+    name: string,
+    type: string,
+    value: string
+}
+
+let g_replVariables: WorkspaceVariable[] = undefined;
+
+export class REPLTreeDataProvider implements vscode.TreeDataProvider<WorkspaceVariable> {
+    private _onDidChangeTreeData: vscode.EventEmitter<WorkspaceVariable | undefined> = new vscode.EventEmitter<WorkspaceVariable | undefined>();
+    readonly onDidChangeTreeData: vscode.Event<WorkspaceVariable | undefined> = this._onDidChangeTreeData.event;
 
     refresh(): void {
         this._onDidChangeTreeData.fire(undefined);
     }
 
-    getChildren(node?: string) {
+    getChildren(node?: WorkspaceVariable) {
         if (node) {
-            return [node]
+            return []
         }
         else {
-            if (g_terminal) {
-                return g_replVariables.split(';').slice(1)
+            if (g_replVariables) {
+                return g_replVariables
             }
             else {
-                return ['no repl attached']
+                return []
             }
         }
     }
 
-    getTreeItem(node: string): vscode.TreeItem {
-        let treeItem: vscode.TreeItem = new vscode.TreeItem(node)
+    getTreeItem(node: WorkspaceVariable): vscode.TreeItem {
+        let treeItem = new vscode.TreeItem(`${node.name}:`)
+        treeItem.description = node.value;
+        treeItem.tooltip = node.type;
+        treeItem.contextValue = 'globalvariable';
         return treeItem;
     }
 }
 
-// TODO Enable again
-// let g_REPLTreeDataProvider: REPLTreeDataProvider = null;
+let g_REPLTreeDataProvider: REPLTreeDataProvider = null;
+
+export async function updateReplVariables() {
+    g_replVariables = await repl.g_connection.sendRequest(repl.requestTypeGetVariables, undefined);
+
+    g_REPLTreeDataProvider.refresh();
+}
```

Comment:
maybe it's better to defined `updateREPLVariables` and `showInVSCode` in repl.ts ? 
We may want to use them in another feature in the future, I guess (e.g. documentation pane).
And it would be more safe to introduce `g_connection` validation check here.

Reply:
Probably yes, but can we leave them in here for now? I wanted to reorganize the code a bit more in general in a follow up PR.

Reply:
yep, of course. I'm welcome to craft that.